pkg/volume/util/nested_volumes.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"k8s.io/api/core/v1"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// MakeNestedMountpoints creates mount points in baseDir for volumes mounted beneath name
+func MakeNestedMountpoints(name, baseDir string, pod v1.Pod) error {
+	for i := 0; i < len(pod.Spec.InitContainers)+len(pod.Spec.Containers); i++ {
+		var container *v1.Container
+		if i < len(pod.Spec.InitContainers) {
+			container = &pod.Spec.InitContainers[i]
+		} else {
+			container = &pod.Spec.Containers[i-len(pod.Spec.InitContainers)]
+		}
+		var mountPoints []string
+		var myMountPoints []string
+		for _, vol := range container.VolumeMounts {
+			mountPoints = append(mountPoints, filepath.Clean(vol.MountPath))
```

Comment:
Use a variable for `filepath.Clean(vol.MountPath)` so you don't need to do `mountPoints[len(mountPoints)-1]` two lines below.